tests/population/test_base_population.py

```diff
@@ -77,16 +78,27 @@ def test_BasePopulation(
     assert mock_args["demographic_proportions"].equals(sub_pop)
     assert mock_args["randomness_streams"] == base_pop.randomness
     pop = simulation.get_population()
-    for column in [col for col in pop.columns if col != "simulant_step_size"]:
+    outer_pop_cols = pop.columns.get_level_values(0)
+    mortality_cols = [col for col in outer_pop_cols if "mortality" in col] + [
+        "alive",
+        "cause_of_death",
+        "years_of_life_lost",
+    ]
+    base_cols = [
+        col
+        for col in outer_pop_cols
+        if col not in mortality_cols + ["simulant_step_size", "is_aged_out"]
+    ]
+    for column in [col for col in base_cols]:
```

Comment:
This seems more complicated than just listing the columns that are created by the `BasePopulation` component. You could also check that these columns match the columns present in `full_simulants`. 

Would something like this work?
```python
assert simulation.get_component("base_population").columns_created == list(full_simulants.columns)
pd.assert_frames_equal(pop[full_simulants.columns], full_simulants)
```